File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.py
T = 10
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    # ///////////////////////////////////////////////////////////////////////////////////
    N = int(input())
    APT = list(map(int, input().split()))
    #       #처음 2제외  97까지
    answer = 0
    for i in range(2, N-2):
        cur = APT[i]
        minus1 = APT[i - 1]
        minus2 = APT[i - 2]
        plus1 = APT[i + 1]
        plus2 = APT[i + 2]
        # 만약 하나라도 그거를 넘게 되면 다음으로 넘어가도 된다.
        # 둘 중에 하나라도 안 넘으면 ok이고 더 큰 값에서 현재 값을 빼서 하면 됨
        temp_max = max(minus1, minus2, plus1, plus2)
        if cur > temp_max:
            answer += cur - temp_max

    print(f'#{test_case} {answer}')

File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.java
import java.util.Scanner;
import java.io.FileInputStream;

class Solution
{
	public static void main(String args[]) throws Exception
	{
		Scanner sc = new Scanner(System.in);
		int T;
		T=10;

		for(int test_case = 1; test_case <= T; test_case++)
		{
			int N = sc.nextInt();
			int[] APT = new int[N];
			for (int i=0;i<N;i++) {
				APT[i] = sc.nextInt();
			}
			int answer = 0;
			
			for (int i = 2; i<N-2; i++) {
			    int cur = APT[i];
			    int minus1 = APT[i - 1];
			    int minus2 = APT[i - 2];
			    int plus1 = APT[i + 1];
			    int plus2 = APT[i + 2];
			    // 만약 하나라도 그거를 넘게 되면 다음으로 넘어가도 된다.
			    // 둘 중에 하나라도 안 넘으면 ok이고 더 큰 값에서 현재 값을 빼서 하면 됨
			    int temp_max = Math.max(minus1, minus2);
			    int temp_min = Math.max(plus1, plus2);
			    int max = Math.max(temp_max, temp_min);
			    if (cur > max) {
			        answer += cur - max;
			    }
			}
			System.out.println("#"+test_case+ " "+answer);
		}
	}
}